### PR TITLE
lib: Port us away from deprecated Select variant "checkbox"

### PR DESCRIPTION
**Pull request by mvollmer**

### CODE REVIEW

File: pkg/lib/cockpit-components-checkbox-select.tsx [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+    const isSelected = selected.includes(value);
+    return (
+      <SelectOption value={value} key={value} hasCheckbox isSelected={isSelected} {...props}>
+        {content}
+      </SelectOption>
+    );
+  });
+
+  const onToggleClick = () => {
+    onToggle && onToggle(!isOpen);
```
Comment (cockpituous) [inline pkg/lib/cockpit-components-checkbox-select.tsx:115]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-checkbox-select.tsx [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    setIsOpen(!isOpen);
+  };
+
+  const _onSelect = (event: React.MouseEvent<Element, MouseEvent> | undefined, value: string | number | undefined) => {
+      if (value && event)
+          onSelect(value, (event.target as HTMLCheckboxElement).checked);
```
Comment (cockpituous) [inline pkg/lib/cockpit-components-checkbox-select.tsx:121]: These 3 added lines are not executed by any test.

File: pkg/lib/cockpit-components-checkbox-select.tsx [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+  );
+
+  return (
+    <Select
+      isOpen={isOpen}
+      selected={selected}
+      onSelect={_onSelect}
+      onOpenChange={(isOpen) => {
+        onToggle && onToggle(isOpen);
+        setIsOpen(isOpen);
```
Comment (cockpituous) [inline pkg/lib/cockpit-components-checkbox-select.tsx:149]: These 3 added lines are not executed by any test.

File: pkg/lib/cockpit-components-simple-select.tsx [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+    const { content, value, ...props } = option;
+    return (
+      <SelectOption value={value} key={value} {...props}>
+        {content}
+      </SelectOption>
+    );
+  });
+
+  const onToggleClick = () => {
+    onToggle && onToggle(!isOpen);
```
Comment (cockpituous) [inline pkg/lib/cockpit-components-simple-select.tsx:127]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-simple-select.tsx [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+    setIsOpen(!isOpen);
+  };
+
+  const _onSelect = (_event: React.MouseEvent<Element, MouseEvent> | undefined, value: string | number | undefined) => {
+    if (value)
+      onSelect(value);
+    onToggle && onToggle(true);
```
Comment (cockpituous) [inline pkg/lib/cockpit-components-simple-select.tsx:134]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-simple-select.tsx [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+    setIsOpen(false);
+  };
+
+  let content: React.ReactNode = placeholder;
+  if (toggleContent)
+    content = toggleContent;
```
Comment (cockpituous) [inline pkg/lib/cockpit-components-simple-select.tsx:140]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-simple-select.tsx [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+  );
+
+  return (
+    <Select
+      isOpen={isOpen}
+      selected={selected}
+      onSelect={_onSelect}
+      onOpenChange={(isOpen) => {
+        onToggle && onToggle(isOpen);
+        setIsOpen(isOpen);
```
Comment (cockpituous) [inline pkg/lib/cockpit-components-simple-select.tsx:168]: These 3 added lines are not executed by any test.

File: pkg/lib/serverTime.js [hunk 8]
```diff
@@ -581,12 +582,11 @@ function ChangeSystimeBody({ state, errors, change }) {
...
-                        isOpen={modeOpen} onToggle={(_, isOpen) => setModeOpen(isOpen)}
-                        selections={mode} onSelect={(event, value) => { setModeOpen(false); change("mode", value) }}
-                        menuAppendTo="parent">
-                    { mode_options }
-                </Select>
+                <SimpleSelect
+                    id="change_systime"
+                    options={mode_options}
+                    selected={mode}
+                    onSelect={value => change("mode", value)} />
```
Comment (cockpituous) [inline pkg/lib/serverTime.js:589]: This added line is not executed by any test.

File: pkg/metrics/metrics.jsx [hunk 9]
```diff
@@ -1632,7 +1634,7 @@ class MetricsHistory extends React.Component {
         this.setState({ isDatepickerOpened: isOpen });
     }
 
-    handleSelect(e, sel) {
+    handleSelect(sel) {
```
Comment (cockpituous) [inline pkg/metrics/metrics.jsx:1637]: This added line is not executed by any test.

File: pkg/metrics/metrics.jsx [hunk 10]
```diff
@@ -1856,52 +1858,49 @@
...
+                                    "aria-label": _("Jump to")
+                                }} />
+                            <CheckboxSelect
+                                toggleProps={{
+                                    "aria-label": _("Graph visibility options menu"),
+                                    className: "select-min metrics-label",
+                                }}
+                                toggleContent={_("Graph visibility")}
+                                noBadge
+                                onSelect={(selection, checked) => {
```
Comment (cockpituous) [inline pkg/metrics/metrics.jsx:1897]: This added line is not executed by any test.

File: pkg/metrics/metrics.jsx [hunk 11]
```diff
@@ -1856,52 +1858,49 @@
...
                                     this.setState(prevState => ({
-                                        selectedVisibility: { ...prevState.selectedVisibility, [s[0]]: !prevState.selectedVisibility[s[0]] }
+                                        selectedVisibility: { ...prevState.selectedVisibility, [selection]: checked }
```
Comment (cockpituous) [inline pkg/metrics/metrics.jsx:1899]: This added line is not executed by any test.

File: pkg/systemd/logs.jsx [hunk 12]
```diff
@@ -167,19 +173,21 @@ export const LogsPage = () => {
         updateUrl(Object.assign(getOptionsFromTextInput(value)));
     };
 
-    const onTimeFilterChange = (newTimeFilter) => {
+    const onTimeFilterChange = (value) => {
```
Comment (cockpituous) [inline pkg/systemd/logs.jsx:176]: This added line is not executed by any test.

File: pkg/systemd/logs.jsx [hunk 13]
```diff
@@ -167,19 +173,21 @@
...
         setUpdateIdentifiersList(true);
 
-        if (newTimeFilter.key == 'boot' && newTimeFilter.value !== "0") // Don't follow if specific boot is picked
+        const newTimeFilter = timeFilterParams[value];
```
Comment (cockpituous) [inline pkg/systemd/logs.jsx:179]: This added line is not executed by any test.

File: pkg/systemd/logs.jsx [hunk 14]
```diff
@@ -167,19 +173,21 @@
...
+
+        if (newTimeFilter.param == 'boot' && newTimeFilter.value !== "0") // Don't follow if specific boot is picked
```
Comment (cockpituous) [inline pkg/systemd/logs.jsx:181]: This added line is not executed by any test.

File: pkg/systemd/logs.jsx [hunk 15]
```diff
@@ -167,19 +173,21 @@
...
             setDataFollowing(false);
-        else if (options.boot && options.boot !== "0" && newTimeFilter.key !== "boot") // Start following is specific boot is removed
+        else if (options.boot && options.boot !== "0" && newTimeFilter.param !== "boot") // Start following is specific boot is removed
```
Comment (cockpituous) [inline pkg/systemd/logs.jsx:183]: This added line is not executed by any test.

File: pkg/systemd/logs.jsx [hunk 16]
```diff
@@ -167,19 +173,21 @@
...
             setDataFollowing(true);
 
         // Remove all parameters which can be set up using filters
         delete options.boot;
         delete options.since;
 
-        cockpit.location.go([], Object.assign(options, { [newTimeFilter.key]: newTimeFilter.value }));
+        cockpit.location.go([], Object.assign(options, { [newTimeFilter.param]: newTimeFilter.value }));
```
Comment (cockpituous) [inline pkg/systemd/logs.jsx:190]: This added line is not executed by any test.

File: pkg/systemd/services.jsx [hunk 17]
```diff
@@ -757,18 +755,17 @@ const ServicesPageFilters = ({
         onOptionsChanged(_options);
     }, [filters, currentTextFilter, onOptionsChanged]);
 
-    const onSelect = (type, event, selection) => {
-        const checked = event.target.checked;
-
+    const onSelect = (type, checked, selection) => {
```
Comment (cockpituous) [inline pkg/systemd/services.jsx:758]: This added line is not executed by any test.

File: pkg/systemd/services.jsx [hunk 18]
```diff
@@ -757,18 +755,17 @@
...
         setFilters({ ...filters, [type]: checked ? [...filters[type], selection] : filters[type].filter(value => value !== selection) });
     };
 
-    const onActiveStateSelect = (event, selection) => {
-        onSelect('activeState', event, selection);
+    const onActiveStateSelect = (selection, checked) => {
+        console.log("ACT", selection, checked);
+        onSelect('activeState', checked, selection);
```
Comment (cockpituous) [inline pkg/systemd/services.jsx:764]: These 3 added lines are not executed by any test.

File: pkg/systemd/services.jsx [hunk 19]
```diff
@@ -757,18 +755,17 @@
...
     };
 
-    const onFileStateSelect = (event, selection) => {
-        onSelect('fileState', event, selection);
+    const onFileStateSelect = (selection, checked) => {
+        onSelect('fileState', checked, selection);
```
Comment (cockpituous) [inline pkg/systemd/services.jsx:768]: These 2 added lines are not executed by any test.

File: pkg/lib/cockpit-components-checkbox-select.tsx (hunk 1 above)
Comment (cockpituous) [inline pkg/lib/cockpit-components-checkbox-select.tsx:115]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-checkbox-select.tsx [hunk 20]
```diff
@@ -0,0 +1,163 @@
...
```
Comment (cockpituous) [inline pkg/lib/cockpit-components-checkbox-select.tsx:149]: These 3 added lines are not executed by any test.

File: pkg/lib/cockpit-components-simple-select.tsx (hunk 4 above)
Comment (cockpituous) [inline pkg/lib/cockpit-components-simple-select.tsx:127]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-simple-select.tsx (hunk 5 above)
Comment (cockpituous) [inline pkg/lib/cockpit-components-simple-select.tsx:134]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-simple-select.tsx (hunk 6 above)
Comment (cockpituous) [inline pkg/lib/cockpit-components-simple-select.tsx:140]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-simple-select.tsx (hunk 7 above)
Comment (cockpituous) [inline pkg/lib/cockpit-components-simple-select.tsx:168]: These 3 added lines are not executed by any test.

File: pkg/metrics/metrics.jsx [hunk 21]
```diff
@@ -1856,52 +1857,46 @@ class MetricsHistory extends React.Component {
...
+                                    "aria-label": _("Jump to")
+                                }} />
+                            <CheckboxSelect
+                                toggleProps={{
+                                    "aria-label": _("Graph visibility options menu"),
+                                    className: "select-min metrics-label",
+                                }}
+                                toggleContent={_("Graph visibility")}
+                                noBadge
+                                onSelect={(selection, checked) => {
```
Comment (cockpituous) [inline pkg/metrics/metrics.jsx:1893]: This added line is not executed by any test.

File: pkg/metrics/metrics.jsx [hunk 22]
```diff
@@ -1856,52 +1857,46 @@
...
                                     this.setState(prevState => ({
-                                        selectedVisibility: { ...prevState.selectedVisibility, [s[0]]: !prevState.selectedVisibility[s[0]] }
+                                        selectedVisibility: { ...prevState.selectedVisibility, [selection]: checked }
```
Comment (cockpituous) [inline pkg/metrics/metrics.jsx:1895]: This added line is not executed by any test.

File: pkg/systemd/logs.jsx [hunk 23]
```diff
@@ -253,7 +253,7 @@ export const LogsPage = () => {
                                                 isDisabled={options.boot && options.boot !== "0"}
                                                 onClick={() => {
                                                     // Reset time filter if following mode is now selected but we are on a specific boot
-                                                    if (!dataFollowing && timeFilter && timeFilter.key == "boot" && timeFilter.value !== "0") {
+                                                    if (!dataFollowing && timeFilterParam && timeFilterParam.param == "boot" && timeFilterParam.value !== "0") {
```
Comment (cockpituous) [inline pkg/systemd/logs.jsx:256]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-checkbox-select.tsx [hunk 24]
```diff
@@ -0,0 +1,175 @@
...
+  noBadge = false,
+  ...props
+}: CheckboxSelectProps) => {
+  const [isOpen, setIsOpen] = React.useState(false);
+
+  const checkboxSelectOptions = options?.map((option) => {
+    const { content, value, key, ...props } = option;
+    const isSelected = selected.includes(value);
+    return (
+      <SelectOption value={value} key={key !== undefined ? key : `${value}`}
```
Comment (cockpituous) [inline pkg/lib/cockpit-components-checkbox-select.tsx:116]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-checkbox-select.tsx [hunk 25]
```diff
@@ -0,0 +1,175 @@
...
+                    hasCheckbox isSelected={isSelected} {...props}>
+        {content}
+      </SelectOption>
+    );
+  });
+
+  const onToggleClick = () => {
+    onToggle && onToggle(!isOpen);
```
Comment (cockpituous) [inline pkg/lib/cockpit-components-checkbox-select.tsx:124]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-checkbox-select.tsx [hunk 26]
```diff
@@ -0,0 +1,175 @@
...
+  );
+
+  return (
+    <Select
+      isOpen={isOpen}
+      selected={selected}
+      onSelect={_onSelect}
+      onOpenChange={(isOpen) => {
+        onToggle && onToggle(isOpen);
+        setIsOpen(isOpen);
```
Comment (cockpituous) [inline pkg/lib/cockpit-components-checkbox-select.tsx:161]: These 3 added lines are not executed by any test.

File: pkg/metrics/metrics.jsx [hunk 27]
```diff
@@ -1885,23 +1883,20 @@ class MetricsHistory extends React.Component {
...
-                                onSelect={(_, selection) => {
-                                    const s = this.columns.find(itm => itm[1] == selection);
+                            <CheckboxSelect
+                                toggleProps={{
+                                    "aria-label": _("Graph visibility options menu"),
+                                    className: "select-min metrics-label",
+                                }}
+                                toggleContent={_("Graph visibility")}
+                                noBadge
+                                onSelect={(selection, checked) => {
```
Comment (cockpituous) [inline pkg/metrics/metrics.jsx:1893]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-checkbox-select.tsx (hunk 25 above)
Comment (cockpituous) [inline pkg/lib/cockpit-components-checkbox-select.tsx:124]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-checkbox-select.tsx [hunk 28]
```diff
@@ -0,0 +1,175 @@
...
+    </MenuToggle>
+  );
+
+  return (
+    <Select
+      isOpen={isOpen}
+      selected={selected}
+      onSelect={_onSelect}
+      onOpenChange={(isOpen) => {
+        onToggle && onToggle(isOpen);
```
Comment (cockpituous) [inline pkg/lib/cockpit-components-checkbox-select.tsx:160]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-checkbox-select.tsx (hunk 24 above)
Comment (cockpituous) [inline pkg/lib/cockpit-components-checkbox-select.tsx:116]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-checkbox-select.tsx (hunk 25 above)
Comment (cockpituous) [inline pkg/lib/cockpit-components-checkbox-select.tsx:124]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-checkbox-select.tsx (hunk 28 above)
Comment (cockpituous) [inline pkg/lib/cockpit-components-checkbox-select.tsx:160]: This added line is not executed by any test.

**Comment by martinpitt** [comment]
@mvollmer removing from my review queue as this is still red. I rather dislike that we have to take on more and more PF code with what should be generic upstream components, but alas.. :man_shrugging: Thanks for working on this!

File: pkg/lib/cockpit-components-checkbox-select.tsx (hunk 24 above)
Comment (cockpituous) [inline pkg/lib/cockpit-components-checkbox-select.tsx:116]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-checkbox-select.tsx (hunk 25 above)
Comment (cockpituous) [inline pkg/lib/cockpit-components-checkbox-select.tsx:124]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-checkbox-select.tsx (hunk 28 above)
Comment (cockpituous) [inline pkg/lib/cockpit-components-checkbox-select.tsx:160]: This added line is not executed by any test.

**Comment by mvollmer** [comment]
I think the remaining failures are flakes, amplified by repeating agffected tests:

 https://cockpit-logs.us-east-1.linodeobjects.com/pull-21373-27a0d5f3-20241219-095700-ubuntu-stable-other/log.html#131

https://cockpit-logs.us-east-1.linodeobjects.com/pull-21373-27a0d5f3-20241219-095702-fedora-coreos-other/log.html#151-2

I can't reproduce them locally...

**Pull request merged**